python/tests/transformers/tf_transformer_test.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import tensorflow as tf
+
+from pyspark.sql.types import Row
+
+import tensorframes as tfs
+
+import sparkdl.graph.utils as tfx
+from sparkdl.graph.input import TFInputGraph
+from sparkdl.transformers.tf_tensor import TFTransformer
+
+from ..tests import SparkDLTestCase
+
+class TFTransformerTests(SparkDLTestCase):
+    def test_graph_novar(self):
+        transformer = _build_transformer(lambda session:
+                                         TFInputGraph.fromGraph(session.graph, session,
+                                                                [_tensor_input_name],
+                                                                [_tensor_output_name]))
+        gin = transformer.getTFInputGraph()
+        local_features = _build_local_features()
+        expected = _get_expected_result(gin, local_features)
+        dataset = self.session.createDataFrame(local_features)
+        _check_transformer_output(transformer, dataset, expected)
+
+
+# The name of the input tensor
+_tensor_input_name = "input_tensor"
+# The name of the output tensor (scalar)
+_tensor_output_name = "output_tensor"
+# The size of the input tensor
+_tensor_size = 3
+# Input mapping for the Transformer
+_input_mapping = {'inputCol': tfx.tensor_name(_tensor_input_name)}
+# Output mapping for the Transformer
+_output_mapping = {tfx.tensor_name(_tensor_output_name): 'outputCol'}
+# Numerical threshold
+_all_close_tolerance = 1e-5
+
+
+def _build_transformer(gin_function):
+    """
+    Makes a session and a default graph, loads the simple graph into it, and then calls
+    gin_function(session) to build the :py:obj:`TFInputGraph` object.
+    Return the :py:obj:`TFTransformer` created from it.
+    """
+    graph = tf.Graph()
+    with tf.Session(graph=graph) as sess, graph.as_default():
+        _build_graph(sess)
+        gin = gin_function(sess)
+
+    return TFTransformer(tfInputGraph=gin,
+                         inputMapping=_input_mapping,
+                         outputMapping=_output_mapping)
+
+
+def _build_graph(sess):
```

Comment:
I deliberately kept the `sess` argument so that if one deems the use of sess necessary in constructing the graph, he or she will only have to to modify it here in this function.